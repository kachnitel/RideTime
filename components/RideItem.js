import React from 'react';
import { Text, View, StyleSheet } from 'react-native';
import { RideItemDetail } from './RideItemDetail';

export class RideItem extends React.Component {
  render() {
    return <View style={styles.ride}>
      <Text style={styles.name}>{this.props.ride.name}</Text>
      <RideItemDetail ride={this.props.ride} />
    </View>;
  }
}

const styles = StyleSheet.create({
    name: {
      padding: 3,
      fontSize: 16,
      height: 26,
    },
    itemCoords: {
      paddingLeft: 5,
      fontSize: 10,
      height: 12,
    },
    ride: {
      height: 64,
      backgroundColor: 'white',
      borderRadius: 4,
      borderWidth: 0.5,
      borderColor: '#d6d7da',
    }
})