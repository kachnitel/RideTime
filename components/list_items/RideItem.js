import React from 'react';
import { Text, View, TouchableHighlight, StyleSheet } from 'react-native';
import { RideItemDetail } from '../RideItemDetail';
import { withNavigation } from 'react-navigation';
import Layout from '../../constants/Layout';

class RideItem extends React.Component {
  render() {
    return (
      <TouchableHighlight onPress={() => this.props.navigation.navigate({
        key: Math.random () * 10000,
        routeName: 'RideDetail',
        params: this.props.ride
      })}>
        <View style={{...styles.listItem, ...this.props.style}}>
          <Text style={{...styles.name, ...this.props.style}}>
            {this.props.ride.name}
          </Text>
          <RideItemDetail
            ride={this.props.ride}
            style={this.props.style}
          />
        </View>
      </TouchableHighlight>
    );
  }
}

export default withNavigation(RideItem);

const styles = StyleSheet.create({
  name: {
    fontSize: Layout.window.hp(2.75),
  },
  listItem: {
    height: Layout.window.hp(15),
    paddingVertical: Layout.window.hp(1.5),
    paddingHorizontal: Layout.window.wp(4)
  }
})
