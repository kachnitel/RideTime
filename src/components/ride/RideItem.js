import React from 'react'
import { View } from 'react-native'
import { observer } from 'mobx-react/native'
import RideItemDetail from './RideItemDetail'
import RidersListCompact from '../ride/RidersListCompact'
import Header from '../Header'

export default
@observer
class RideItem extends React.Component {
  render () {
    return (
      <View>
        <Header>{this.props.ride.title}</Header>
        <RideItemDetail ride={this.props.ride} />
        <RidersListCompact userIDs={this.props.ride.members.slice()} />
      </View>
    )
  }
}
