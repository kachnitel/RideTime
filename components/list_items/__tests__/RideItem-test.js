/* eslint-env jest */
import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import RideItem from '../RideItem'

jest.mock('../../RideItemDetail', () => 'RideItemDetail')

let testRide = {
  'name': "Joey's Alice lake lap",
  'difficulty': 1,
  'location': 'Alice Lake',
  'locationGps': [49.775666, -123.123307],
  'members': [1],
  'terrain': 'trail',
  'route': 'Some trails'
}

it('renders correctly', () => {
  const tree = renderer.create(<RideItem ride={testRide} />).toJSON()

  expect(tree).toMatchSnapshot()
})

it('renders correctly with style', () => {
  const tree = renderer.create(<RideItem
    ride={testRide}
    style={{
      backgroundColor: 'black',
      color: '#fff'
    }}
  />).toJSON()

  expect(tree).toMatchSnapshot()
})
