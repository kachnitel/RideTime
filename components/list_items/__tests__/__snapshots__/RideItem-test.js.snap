// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "height": 200,
      "paddingHorizontal": 30,
      "paddingVertical": 20,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 36.5,
      }
    }
  >
    Joey's Alice lake lap
  </Text>
  <RideItemDetail
    ride={
      Object {
        "difficulty": 1,
        "location": "Alice Lake",
        "locationGps": Array [
          49.775666,
          -123.123307,
        ],
        "members": Array [
          1,
        ],
        "name": "Joey's Alice lake lap",
        "plannedRoute": "Some trails",
        "terrain": "trail",
      }
    }
  />
</View>
`;

exports[`renders correctly with style 1`] = `
<View
  style={
    Object {
      "backgroundColor": "black",
      "color": "#fff",
      "height": 200,
      "paddingHorizontal": 30,
      "paddingVertical": 20,
    }
  }
>
  <Text
    style={
      Object {
        "backgroundColor": "black",
        "color": "#fff",
        "fontSize": 36.5,
      }
    }
  >
    Joey's Alice lake lap
  </Text>
  <RideItemDetail
    ride={
      Object {
        "difficulty": 1,
        "location": "Alice Lake",
        "locationGps": Array [
          49.775666,
          -123.123307,
        ],
        "members": Array [
          1,
        ],
        "name": "Joey's Alice lake lap",
        "plannedRoute": "Some trails",
        "terrain": "trail",
      }
    }
    style={
      Object {
        "backgroundColor": "black",
        "color": "#fff",
      }
    }
  />
</View>
`;
